### PR TITLE
Fix logic issue with garage door. 

### PR DESCRIPTION
**Pull request by gwaland**
In home assistant if the garage door is closed and you call cover.close on the entity it will open the garage.  The expected functionality is that nothing will happen as the garage is already closed. 

In home assistant if the garage door is open and you call cover.open on the entity it will close the garage.  The expected functionality is that nothing will happen as the garage is already open. 

This pull request adds a check in each command to make this function as expected.

### CODE REVIEW

File: athom-garage-door.yaml [hunk 1]
```diff
@@ -99,11 +99,21 @@ cover:
     name: "${friendly_name}"
     lambda: "return id(contact).state ? COVER_OPEN : COVER_CLOSED;"
     open_action:
-      - switch.turn_on: relay
+      then:
+        - if:
+            condition:
+              lambda: 'return id(contact).state == false;'
```
Comment (jesserockz) [inline athom-garage-door.yaml:105]:
```suggestion
              lambda: 'return !id(contact).state;'
```

File: athom-garage-door.yaml [hunk 2]
```diff
@@ -99,11 +99,21 @@ cover:
...
+            then:
+              - switch.turn_on: relay
     stop_action:
       - switch.turn_on: relay
     close_action:
-      - switch.turn_on: relay
+      then:
+        - if:
+            condition:
+              lambda: 'return id(contact).state == true;'
```
Comment (jesserockz) [inline athom-garage-door.yaml:114]:
```suggestion
              lambda: 'return id(contact).state;'
```

**Review by jesserockz — CHANGES_REQUESTED** [review]

File: athom-garage-door.yaml (hunk 1 above)
Comment (gwaland) [reply athom-garage-door.yaml]: Adjustments added

**Comment by gwaland** [comment]
I'm confused about the ci build fail on this?  I didn't submit this 35 days ago.

**Pull request merged**